### PR TITLE
[Fix](inverted index) fix bug when match condition in hash join

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

When SQL query like this:
```
SELECT 
    t1.web_name, 
    COUNT(t1.web_name) AS count 
FROM 
    web_site t1 
LEFT JOIN 
    web_sales t2 ON t1.web_site_sk = t2.ws_web_site_sk 
WHERE 
    t1.web_name MATCH 'site_0' 
    OR t2.ws_item_sk > 17934 
GROUP BY 
    t1.web_name 
ORDER BY 
    t1.web_name;
```
match predicate can not push down to scan node, it stays in hash join node,  in nereids planner, when visitHashJoinNode, olap table schema can not be got from the slotRef, which is intermediate, which caused inverted index context can not get correctly.

In this PR, we push olap table to intermediate slotRef's tableIf, so match predicate can get the right inverted index context.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.7 seconds
 stream load tsv:          534 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17161897803 Bytes

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java [hunk 1]
```diff
@@ -169,38 +169,61 @@ public Expr visitLessThanEqual(LessThanEqual lessThanEqual, PlanTranslatorContex
...
+        OlapTable olapTbl = Optional.ofNullable(getOlapTableFromSlotDesc(left.getDesc()))
+                                    .orElse(getOlapTableDirectly(left));
+
+        if (olapTbl == null) {
+            throw new AnalysisException("slotRef in matchExpression failed to get OlapTable");
+        }
+
+        List<Index> indexes = olapTbl.getIndexes();
+        if (indexes == null) {
+            return null; // or throw an exception if this is unexpected
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java:205]: should not return `null` ever

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java [hunk 2]
```diff
@@ -169,38 +169,61 @@ public Expr visitLessThanEqual(LessThanEqual lessThanEqual, PlanTranslatorContex
...
+
+        if (left == null) {
+            throw new AnalysisException("Left slot reference is null");
+        }
+        OlapTable olapTbl = Optional.ofNullable(getOlapTableFromSlotDesc(left.getDesc()))
+                                    .orElse(getOlapTableDirectly(left));
+
+        if (olapTbl == null) {
+            throw new AnalysisException("slotRef in matchExpression failed to get OlapTable");
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java:201]: if the left expr must a slot reference from table, u should check it in `CheckMatchExpression` rule

Comment (airborne12) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java:201]: > if the left expr must a slot reference from table, u should check it in `CheckMatchExpression` rule

already checked, as in the following code
` private Plan checkChildren(LogicalFilter<? extends Plan> filter) {
        List<Expression> expressions = filter.getExpressions();
        for (Expression expr : expressions) {
            if (expr instanceof Match) {
                Match matchExpression = (Match) expr;
                if (!(matchExpression.left() instanceof SlotReference)
                        || !(matchExpression.right() instanceof Literal)) {
                    throw new AnalysisException(String.format("Only support match left operand is SlotRef,"
                            + " right operand is Literal. But meet expression %s", matchExpression));
                }
            }
        }
        return filter;
    }`

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.67 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17161992476 Bytes

**Comment by airborne12** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.73 seconds
 stream load tsv:          551 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162128441 Bytes

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by starocean999 — APPROVED** [review]

**Pull request merged**